OpenRA.Mods.Common/Widgets/ButtonWidget.cs

```diff
@@ -243,12 +243,27 @@ public override void Draw()
 			var bgDark = GetContrastColorDark();
 			var bgLight = GetContrastColorLight();
 
+			var contentWidth = Bounds.Width - LeftMargin - RightMargin;
 			var stateOffset = Depressed ? new int2(VisualHeight, VisualHeight) : new int2(0, 0);
 
 			var position = GetTextPosition(text, font, rb);
 
 			var hover = Ui.MouseOverWidget == this || Children.Any(c => c == Ui.MouseOverWidget);
 			DrawBackground(rb, disabled, Depressed, hover, highlighted);
+
+			// Scissor when the text overflows
+			var isTextOverflowing = font.Measure(text).X > contentWidth;
+			if (isTextOverflowing && TooltipContainer != null)
+			{
+				Game.Renderer.EnableScissor(new Rectangle(RenderOrigin.X + LeftMargin, RenderOrigin.Y,
```

Comment:
Why is this scissoring *and* truncating the text? The scissoring will be redundant here.

Reply:
You mean that truncating is sufficient? I didn't realize that... At first it was only scissoring (modeled after hotkey input/text input widget) but there had to be a visual cue so I added truncation. I can fix that in a follow up.

Reply:
`TruncateText` guarantees that the text width is less than `contentWidth`, provided that `contentWidth` is wide enough to fit at least "...".